aleph-client/src/contract/event.rs

```diff
@@ -0,0 +1,220 @@
+//! Utilities for listening for contract events.
+//!
+//! To use the module you will need to first create a subscription (a glorified `Receiver<String>`),
+//! then run the listen loop. You might want to run the loop in a separate thread.
+//!
+//! ```no_run
+//! # use std::sync::mpsc::channel;
+//! # use std::thread;
+//! # use std::time::Duration;
+//! # use aleph_client::{Connection};
+//! # use aleph_client::contract::ContractInstance;
+//! # use aleph_client::contract::event::{listen_contract_events, subscribe_events};
+//! # use anyhow::Result;
+//! # use sp_core::crypto::AccountId32;
+//! # fn example(conn: Connection, address1: AccountId32, address2: AccountId32, path1: &str, path2: &str) -> Result<()> {
+//!     let subscription = subscribe_events(&conn)?;
+//!     let contract1 = ContractInstance::new(address1, path1)?;
+//!     let contract2 = ContractInstance::new(address2, path2)?;
+//!     let (cancel_tx, cancel_rx) = channel();
+//!
+//!     thread::spawn(move || {
+//!         listen_contract_events(
+//!             subscription,
+//!             &[&contract1, &contract2],
+//!             Some(cancel_rx),
+//!             |event_or_error| { println!("{:?}", event_or_error) }
+//!         );
+//!     });
+//!
+//!     thread::sleep(Duration::from_secs(20));
+//!     cancel_tx.send(()).unwrap();
```

Comment:
I would encourage you to write a new helper method that gets a collection of `&ContractInstance`, connection, event handler and timeout and does: subscription, spawning and utilizing `cancel_tx` as a timeout